pkg/operator/controller/ingress_status.go

```diff
@@ -7,12 +7,145 @@ import (
+
+	toolscache "k8s.io/client-go/tools/cache"
+
 	operatorv1 "github.com/openshift/api/operator/v1"
 
 	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+type ingressStatus struct {
+	controllers toolscache.Indexer
+	services    toolscache.Indexer
+}
+
+func isProvisioned(service *corev1.Service) bool {
+	ingresses := service.Status.LoadBalancer.Ingress
+	return len(ingresses) > 0 && (len(ingresses[0].Hostname) > 0 || len(ingresses[0].IP) > 0)
+}
+
+func isPending(service *corev1.Service) bool {
+	return !isProvisioned(service)
+}
+
+func controlledBy(service *corev1.Service, matches func(*corev1.Service) bool) []string {
+	if service.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		return []string{}
+	}
+	if !matches(service) {
+		return []string{}
+	}
+	controller, ok := service.Labels[manifests.OwningIngressControllerLabel]
+	if !ok {
+		return []string{}
+	}
+	return []string{controller}
+}
+
+func indexByPublishing(obj interface{}) ([]string, error) {
+	c, ok := obj.(*operatorv1.IngressController)
+	if !ok {
+		return []string{}, nil
+	}
+	return []string{string(c.Status.EndpointPublishingStrategy.Type)}, nil
+}
+
+func indexByProvisioned(obj interface{}) ([]string, error) {
+	service, ok := obj.(*corev1.Service)
+	if !ok {
+		return []string{}, nil
+	}
+	return controlledBy(service, isProvisioned), nil
+}
+
+func indexByPending(obj interface{}) ([]string, error) {
+	service, ok := obj.(*corev1.Service)
+	if !ok {
+		return []string{}, nil
+	}
+	return controlledBy(service, isPending), nil
+}
+
+func indexContains(indexer toolscache.Indexer, name, value string) bool {
+	objects, err := indexer.ByIndex(name, value)
+	if err != nil {
+		return false
+	}
+	return len(objects) > 0
+}
+
+var serviceIndexers = toolscache.Indexers{
+	"pending-for":     indexByPending,
+	"provisioned-for": indexByProvisioned,
+}
+var controllerIndexers = toolscache.Indexers{
+	"publishing": indexByPublishing,
+}
+
+func NewIngressStatus(controllers, services toolscache.Indexer) *ingressStatus {
+	controllers.AddIndexers(controllerIndexers)
+	services.AddIndexers(serviceIndexers)
+	return &ingressStatus{
+		controllers: controllers,
+		services:    services,
+	}
+}
+
+func (s *ingressStatus) computeLoadBalancerStatus(ic *operatorv1.IngressController) []operatorv1.OperatorCondition {
+	if ic.Status.EndpointPublishingStrategy.Type != operatorv1.LoadBalancerServiceStrategyType {
+		return []operatorv1.OperatorCondition{
+			{
+				Type:    "LoadBalancerManaged",
+				Status:  operatorv1.ConditionFalse,
+				Reason:  "UnsupportedEndpointPublishingStrategy",
+				Message: fmt.Sprintf("The %s endpoint publishing strategy does not support a load balancer", ic.Status.EndpointPublishingStrategy.Type),
+			},
+		}
+	}
+
+	conditions := []operatorv1.OperatorCondition{}
+
+	conditions = append(conditions, operatorv1.OperatorCondition{
+		Type:    "LoadBalancerManaged",
```

Comment:
`"LoadBalancerManaged"`→`operatorv1.LoadBalancerManagedIngressConditionType`

Reply:
Switched to correct values, deferring upgrade of client-go for now.